pkg/limayaml/validate.go

```diff
@@ -641,3 +655,36 @@ func ValidateAgainstLatestConfig(yNew, yLatest []byte) error {
 
 	return nil
 }
+
+func validateSocket(engine, socket string) error {
+	if socket == "" {
+		return fmt.Errorf("%s socket path must not be empty", engine)
+	}
+
+	u, err := url.Parse(socket)
+	if err == nil && u.Scheme != "" {
+		switch u.Scheme {
```

Comment:
You are ignoring any error from `url.Parse()` here.

```suggestion
	if err != nil {
		return fmt.Errorf("field `%s` %q cannot be parsed as a URL: %w", field, err)
	}
	if u.Scheme == "" {
		// ??? check that the path is absolute?
		return nil
	}
	switch u.Scheme {
```